prov/efa/src/efa_msg.c

```diff
@@ -169,7 +178,9 @@ static ssize_t efa_post_recv(struct efa_ep *ep, const struct fi_msg *msg, uint64
 	if (flags & FI_MORE)
 		return 0;
 
-	err = ibv_post_recv(qp->ibv_qp, ep->recv_more_wr_head.next, &bad_wr);
+	err = -ibv_post_recv(qp->ibv_qp, ep->recv_more_wr_head.next, &bad_wr);
+	if (err == -FI_ENOMEM)
```

Comment:
This is wrong.  ibv_post_recv returns an ERRNO value, not a Libfabric error value.  You should just compare against ENOMEM here.

Reply:
fixed in new revision.